Update SecurityProviderInterface.adoc to use Jakarta terms

xstefank opened the pull request:
Signed-off-by: xstefank <xstefank122@gmail.com>

kazumura commented on specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc:
```diff
@@ -1,64 +1,62 @@
+include some or all of these facilities, as described in link:#a3212[See Profiles]”.
 
-=== Java™ EE Connector Architecture
+=== Jakarta™ EE Connectors
```
should be  Jakarta™ Connectors

xstefank replied:
Thanks. Fixed.

kwsutter commented on specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc:
```diff
@@ -1,64 +1,62 @@
+between a Jakarta EE container and an authorization policy provider.
 
-The JACC specification is available at
+The Jakarta Authorization specification is available at
```
This link is actually for the old JACC specification...  Since we don't have the updated Jakarta Authorization Spec yet, we should probably revert this change to explain that the original JACC spec is available here...  Either that or just remove this reference in this portion of the doc.

kwsutter commented on specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc:
```diff
@@ -1,64 +1,62 @@
 transaction for rollback.
 
-The JTA specification is available at
+The Jakarta Transactions specification is available at
```
Same comment here...  This link is the Java EE Transactions doc, not the Jakarta Spec...  We should have a single reference section at the end of the doc with references to the old and new Specs.  Maybe we should just be pointing at that?  Or, just remove this explicit reference.

xstefank replied:
I agree. Can you point me to the new spec documents, please? I can't find them.

kwsutter replied:
We don't have the "standard" location for the updates specs yet...  At this point, let's just be consistent.  If we want to reference the JCP spec, then that's okay.  But, we shouldn't talk about Jakarta Spec and then point at the JCP location.

kwsutter commented on specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc:
```diff
@@ -1,64 +1,62 @@
+plugged into the Jakarta Persistence framework.
 
-The Java Persistence specification is available
+The Jakarta Persistence specification is available
```
Same comment again...

kwsutter commented on specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc:
```diff
@@ -1,64 +1,62 @@
 mailbox formats.
 
-The JavaMail specification is available at
+The Jakarta Mail specification is available at
```
And, again...

xstefank commented:
@kwsutter I moved JCP links to the end of the documents and added section also for new Jakarta spec documents. I will make a note to update this section when the documents will be available.

xstefank commented:
@kwsutter can you take a long again, please?

kwsutter commented:
@xstefank Thanks, Martin.  That's a good compromise by moving the references to the end of the section.  If or when we have the proper specs for Jakarta, we can update this section.  Thanks.  I will approve and merge.

xstefank commented:
Thanks @kwsutter!

bshannon commented on specification/src/main/asciidoc/platform/ServiceProviderInterface.adoc:
```diff
@@ -1,64 +1,61 @@
-_http://jcp.org/en/jsr/detail?id=919_ .
+=== Specification documents
+
+==== Original Java™ EE specification documents
```
It will be cleaner and less confusing if we don't reference the JCP specs at all.
It would be better to omit these sections at the end and just depend on the
appendix that will link to all the specification documents.

xstefank replied:
@bshannon ok, but I believe there are more sections that reference original JCP links. It would be better to do this in one PR for all documents after the new spec links are available.

bshannon replied:
The expected form of spec links is described in [this message from Wayne](https://www.eclipse.org/lists/jakarta.ee-spec/msg00035.html).

xstefank replied:
Would then make sense to update it right away? I saw the message after the PR was merged.
